tamr_unify_client/client.py

```diff
@@ -140,3 +140,15 @@ def datasets(self):
         :rtype: :class:`~tamr_unify_client.models.DatasetCollection`
         """
         return self._datasets
+
+    def __repr__(self):
+        # intentionally leaving out auth in order to avoid security concerns
+        return (
+            f"{self.__class__.__module__}."
+            f"{self.__class__.__qualname__}("
```

Comment:
Thinking out loud:

We want users to import the client via `from tamr_unify_client import Client`. AFAIK `__qualname__` is intentionally designed to not be smart about re-finagling imports (e.g. [__init__.py](https://github.com/Datatamer/unify-client-python/blob/master/tamr_unify_client/__init__.py)).

That leads to the import being differently namespaced than the qualname.

One thing to consider: at some point we might want to re-finagle more imports in `__init__.py` to make importing easier (less unnecessary namespacing for users). If we do that, then getting rid of `__qualname__` here would make things consistent between the import and the repr.

---

Action: Let's keep this as is.

Reply:
`__qualname__` is only the class name, relative to it's module. In this case:
- `__qualname__` => Client
- `__module__` => tamr\_unify\_client.client

Certain classes could be manually "re-homed" if we want to show (for example) "tamr\_unify\_client" rather than "tamr\_unify\_client.client". This would need to be hard-coded.

> Action: Let's keep this as is.

That's the conclusion I reach as well. The simplest option that still provides consistency and clarity is to use the classes actual module, rather than re-homing.

Also, worth noting that the existing default `__repr__` of this object shows the full path as well. So it's at least not a departure from existing behavior. (Which leaves the door open for a different decision later, if desired)